### PR TITLE
Add CLI E2E tests for MySQL

### PR DESCRIPTION
**Pull request by nahuelverdugo**
### Describe your changes :
- Add CLI E2E tests for MySQL that run every day at midnight. 

### Type of change :
- [x] Improvement
- [x] New feature

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
Ingestion: @open-metadata/ingestion

### CODE REVIEW

File: .github/workflows/py-tests.yml [hunk 1]
```diff
@@ -76,3 +76,29 @@ jobs:
       run: |
         source env/bin/activate
         make run_python_tests
+
+      # we have to pass these args values since we are working with the 'pull_request_target' trigger
+    - name: Push Results in PR to Sonar
```
Comment (pmbrull) [inline .github/workflows/py-tests.yml:81]: thanks!!!

File: ingestion/src/metadata/ingestion/source/database/common_db_source.py [hunk 2]
```diff
@@ -450,6 +449,7 @@ def connection(self) -> Connection:
     def close(self):
         if self.connection is not None:
             self.connection.close()
+        self.engine.dispose()
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/common_db_source.py:452]: good catch

File: ingestion/tests/cli_e2e/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- It partially implements some methods from `test_cli_db_base`.
+- `test_cli_{connector}` is the specific connector test. More tests apart the ones implemented by the `test_cli_db_base` can be run inside this class.
+
+## How to add a database connector
+
+1. Use `test_cli_mysql.py` as example. Your connector E2E CLI test must follow the name convention: `test_cli_{connector}.py` and the test 
+class must extend from `CliCommonDB.TestSuite` if the connector's source implement the `CommonDbSourceService` class, otherwise, from `CliDBBase.TestSuite`.
+
+2. Add an ingestion YAML file with the service and the credentials of it. Use when possible a Dockerized environment, otherwise, remember to use environment
+variables for sensitive information in case of external resources.
```
Comment (pmbrull) [inline ingestion/tests/cli_e2e/README.md:16]: Maybe we can add a line about how the different YAMLs get generated automatically for each workflow run

File: ingestion/tests/cli_e2e/test_cli_db_base_common.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+from sqlalchemy.engine import Engine
+
+from metadata.ingestion.api.sink import SinkStatus
+from metadata.ingestion.api.source import SourceStatus
+from metadata.ingestion.api.workflow import Workflow
+
+from .test_cli_db_base import PATH_TO_RESOURCES, CliDBBase
+
+
+class CliCommonDB(object):
```
Comment (pmbrull) [inline ingestion/tests/cli_e2e/test_cli_db_base_common.py:27]: let's kill the `object` inheritance here, not needed at all

Comment (pmbrull) [reply ingestion/tests/cli_e2e/test_cli_db_base_common.py:27]: Also maybe silly question, I see that we are creating classes as `class MysqlCliTest(CliCommonDB.TestSuite):`. What's the benefit of wrapping the class with `CliCommonDB`? To not trigger the tests?

Maybe we can then have a function returning the `TestSuite` or not use `test_` as a module name 🤔

Comment (nahuelverdugo) [reply ingestion/tests/cli_e2e/test_cli_db_base_common.py:27]: Using this approach we can create an abstract `TestCase` class that avoids issues with test runner discovery and let us still import the abstract test from another module. I had already used this approach in the past for the Secrets Manager tests.

Comment (pmbrull) [reply ingestion/tests/cli_e2e/test_cli_db_base_common.py:27]: cool thanks for explaining. Let's then just remove the `object` notation and call it a day!

**Review by pmbrull — APPROVED** [review]

**Pull request merged**